### DISCUSSION TITLE
Migrate from Gluu-server 4.2

### DISCUSSION THREAD

**Discussion by BainMcKay** [JanssenProject/jans#6923] [Q&A]
We use GLUU-server 4.2  It has an html dashboard UI, which we have customized. 
1. Is that code no longer available? 
2. If not, does Jansen have the same HTML UI, or is it replaced by TUI?
3. How do we go from gluu-server 4.2 to gluu-server 4.3 or 4.4?

**Comment by mzico** [comment]
1. Gluu 4 Source code (nor binaries) are available under Apache 2.0 license and always available here: https://github.com/GluuFederation (for example, see oxAuth, oxTrust, community-edition-setup repos).  
2. Janssen has interactive text-ui (called the "TUI"). If you want web UI, there is a Admin UI component in [Gluu Flex](https://gluu.org/flex), which is Gluu's commercial distribution of the Janssen stack.
3. See [Gluu docs](https://gluu.org/docs/gluu-server/4.5/upgrade/). See also: [Gluu 4](https://gluu.org/gluu-4/)

**Comment by BainMcKay** [comment]
I am now connecting from Wordpress Daggerhart OIDC client wrapper to Janssen. All parameters look good: client ID ,client Secret, urls, et al. I submit the rest api call for [.... /jans-auth/jans-auth/restv1/authorize?...] and I get a 404 (not found),. The site adaptor shows no errors. It should be routing on location as follows:

`<Location /jans-auth>`
`       ProxyPass http://localhost:8081/jans-auth retry=5 connectiontimeout=60 timeout=60`
`        Header set Access-Control-Allow-Origin "*"`
`       Order deny,allow`

CORS should be handled. 

Do I need a whitelist?

**Comment by BainMcKay** [comment]
Here is the header 

`scheme`
`	https`
`host`
`	[FQDN]`
`filename`
`	/jans-auth/restv1/authorize`
`response_type`
`	code`
`scope`
`	email profile openid`
`client_id`
`	[CLIENTID]`
`state`
`	[CLIENTSECRET]`
`redirect_uri`
`	https://[FQDN]/wp-admin/admin-ajax.php?action=openid-connect-authorize`

`Status.                    404 Not Found`
`Version                    HTTP/1.1`
`Transferred             93.04 kB (92.47 kB size)`
`Referrer Policy       strict-origin-when-cross-origin`
`Request Priority     Highest`
`DNS Resolution.     System`

**Reply by BainMcKay** [reply]
Here is the response in a CURL

curl -i -v -L https://[FQDN]/jans-auth/restv1/authorize?response_type=code&scope=email%20profile%20openid&client_id=7092cab5-a883-4db4-b29f-387884460015&state=241760131d1ca5a9c4cc3e27e3ed7716&redirect_uri=https%3A%2F%2F[FQDN]%2Fwp-admin%2Fadmin-ajax.php%3Faction%3Dopenid-connect-authorize
`[1] 2390927`
`[2] 2390928`
`[3] 2390929`
`[4] 2390930`
`[2]   Done                    scope=email%20profile%20openid`
`[3]-  Done                    client_id=[CLIENT)ID]`
`/var/www/html/wordpress# *  Trying [IP_ADDRESS]:443...`
`*TCP_NODELAY set`
`*Connected to c[FQDN] ([IP_ADDRESS]) port 443 (#0)`
`*ALPN, offering h2`
`*ALPN, offering http/1.1`
`*successfully set certificate verify locations:`
`*  CAfile: /etc/ssl/certs/ca-certificates.crt`
  CApath: /etc/ssl/certs`
`*TLSv1.3 (OUT), TLS handshake, Client hello (1):`
`*TLSv1.3 (IN), TLS handshake, Server hello (2):`
`*TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):`
`*TLSv1.3 (IN), TLS handshake, Certificate (11):`
`*TLSv1.3 (IN), TLS handshake, CERT verify (15):`
`*TLSv1.3 (IN), TLS handshake, Finished (20):`
`*TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):`
`*TLSv1.3 (OUT), TLS handshake, Finished (20):`
`*SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384`
`*ALPN, server accepted to use http/1.1`
`*Server certificate:`
`* subject: CN=[FQDN]`
`* start date: Dec 15 18:26:00 2023 GMT`
`* expire date: Mar 14 18:25:59 2024 GMT`
`* sub"2023 04:51:36 GMT`
`Date: Sun, 24 Dec 2023 04:51:36 GMT`
`< Server: Apache/2.4.41 (Ubuntu)`
`Server: Apache/2.4.41 (Ubuntu)`
`< Set-Cookie: PHPSESSID=61bjg6clll6aan1700t0anjn6n; path=/`
`Set-Cookie: PHPSESSID=61bjg6clll6aan1700t0anjn6n; path=/`
`< Expires: Wed, 11 Jan 1984 05:00:00 GMT`
`Expires: Wed, 11 Jan 1984 05:00:00 GMT`
`< Cache-Control: no-cache, must-revalidate, max-age=0`
`Cache-Control: no-cache, must-revalidate, max-age=0`
`< Pragma: no-cache`
`Pragma: no-cache`
`< Link: <https://[FQDN]/wp-json/>; rel="https://api.w.org/"`
'Link: <https://[FQDN]/wp-json/>; rel="https://api.w.org/"`
`< X-TEC-API-VERSION: v1`
`X-TEC-API-VERSION: v1`
`< X-TEC-API-ROOT: https://[FQDN]/wp-json/tribe/events/v1/`
`X-TEC-API-ROOT: https://[FQDN]/wp-json/tribe/events/v1/`
`< X-TEC-API-ORIGIN: https://[FQDN]`
`X-TEC-API-ORIGIN: https://[FQDN]`
`< Transfer-Encoding: chunked`
`Transfer-Encoding: chunked`
`< Content-Type: text/html; charset=UTF-8`
`Content-Type: text/html; charset=UTF-8`
`< `
`<!DOCTYPE html>`
`<html lang="en-US">`
`<head>`
`	<meta charset="UTF-8">`
`	<meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />`
`		<meta name='robots' content='noindex, follow' />`
``
`	<!-- This site is optimized with the Yoast SEO plugin v16.1.1 - https://yoast.com/wordpress/plugins/seo/ -->`
`	<title>Page not found - [TITLE]</title>`
`	<meta property="og:locale" content="en_US" />`
``
`...rest of page with 404 error message'

**Reply by BainMcKay** [reply]
Everything is working now. All the mapping. the logging with authentication. Final issue to resolve is the message in place of the targeted app info screen:

`Application not configured.`

I am working on that now.

**Reply by BainMcKay** [reply]
I got it Woking .  The above issue was another bad (corrupt) plugin.  It's fully functional now.  So Wordpress is done. I am across the line. The rest should be relatively straight forward.  Next are Moodle, NextCloud, Superset and CKAN to get the Knowledge Hub up and running - at GLUU 4.2.2 connectivity and of course  beyond. I can say, that once you figure out the mapping, Janssen is solid. There was some customer pushback on no Web Admin UI, but we will create in a Wordpress Admin Dashboard. Then connect to Janssen on the server through an encrypted AJAX Rest API call a Text Based transaction pipeline so you remove the attack surface. 

I will add detail to the documentation so the oxauth path is easy to cross map from Gluuu tpoJanssen.

**Comment by BainMcKay** [comment]
Made great progress. Allvthe mapping is powering now. It authenticates. Finally message after login is 
`Application not configured.`

Working on that now.
